driver/vhci/vhci_read.c

```diff
@@ -289,7 +317,7 @@ store_urb_class_vendor(PIRP irp, PURB urb, struct urb_req *urbr)
 
 	irp->IoStatus.Information = sizeof(struct usbip_header);
 
-	if (!in) {
+	if (!in && get_read_payload_length(irp) > 0) {
```

Comment:
Please revert to old code. It would be better to focus on GET_STATUS feature. Let us consider empty transfer case later.